src/main/java/com/salesforce/cdp/queryservice/core/QueryServiceConnection.java

```diff
@@ -135,6 +147,10 @@ public QueryEngineEnum getQueryEngineEnum() {
         return queryEngineEnum;
     }
 
+    public @NotNull MetadataCacheKey getMetadataCacheKey() {
+        return new MetadataCacheKey(token.getInstanceUrl(), (String) getDataspace());
+    }
```

Comment:
dataspace variable is in QueryServiceConnection already, right?

Reply:
@soaggarwal I changed the implementation here. `dataspace` variable is not needed since we are not going to change the dataspace after the connection is created, but there is one place I saw where the data space is set explicitly in connection here: refer getTables in QueryServiceMetaData class. @vish689 confirmed that's a different bug we need to fix it in future PRs, so for now, I have kept the cache key as tenantUrl + metadata path. Let me know if this doesn't make sense.